scrapinghub/client.py

```diff
@@ -0,0 +1,370 @@
+                for method in [method[0] for method in methods]:
+                    wrapped = wrap_kwargs(getattr(self, method), apply_fn)
+                    setattr(self, method, wrapped)
+
+    def _proxy_methods(self, methods):
+        proxy_methods(self._origin, self, methods)
+
+
+class Logs(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Logs, self).__init__(*args, **kwargs)
+        self._proxy_methods(['log', 'debug', 'info', 'warning', 'warn',
+                             'error', 'batch_write_start'])
+
+    def _modify_iter_filters(self, params):
+        offset = params.pop('offset', None)
+        if offset:
+            params['start'] = '{}/{}'.format(self.key, offset)
+        level = params.pop('level', None)
+        if level:
+            minlevel = getattr(LogLevel, level, None)
+            if minlevel is None:
+                raise APIError("Unknown log level: {}".format(level))
+            params['filter'] = json.dumps(['level', '>=', [minlevel]])
+        return params
+
+
+class Items(_Proxy):
+
+    def _modify_iter_filters(self, params):
+        offset = params.pop('offset', None)
+        if offset:
+            params['start'] = '{}/{}'.format(self.key, offset)
+        return params
+
+
+class Requests(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Requests, self).__init__(*args, **kwargs)
+        self._proxy_methods(['add'])
+
+
+class Samples(_Proxy):
+    pass
+
+
+class Activity(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Activity, self).__init__(*args, **kwargs)
+        self._proxy_methods([('iter', 'list')])
+
+    def add(self, *args, **kwargs):
+        self._origin.add(*args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        self._origin.post(*args, **kwargs)
+
+
+class Collections(_Proxy):
+
+    def __init__(self, *args, **kwargs):
+        super(Collections, self).__init__(*args, **kwargs)
+        self._proxy_methods([
+            'count', 'get', 'set', 'delete', 'create_writer',
+            '_validate_collection',
+        ])
+
+    def new_store(self, colname):
+        return self.new_collection('s', colname)
+
+    def new_cached_store(self, colname):
+        return self.new_collection('cs', colname)
+
+    def new_versioned_store(self, colname):
+        return self.new_collection('vs', colname)
+
+    def new_versioned_cached_store(self, colname):
+        return self.new_collection('vcs', colname)
+
+    def new_collection(self, coltype, colname):
+        self._validate_collection(coltype, colname)
+        return Collection(self._client, self, coltype, colname)
+
+
+class Collection(object):
+
+    def __init__(self, client, collections, coltype, colname):
+        self._client = client
+        self._origin = _Collection(coltype, colname, collections._origin)
+        proxy_methods(self._origin, self, [
+            'create_writer', 'get', 'set', 'delete', 'count',
```

Comment:
one note about `get` method - in Hubstorage client if you don't pass key to get - it returns an iterator over the entire collection. literally the same thing as `iter`. I think it makes more sense to make `key` parameter required for `get` method.